### ISSUE TITLE
how to write batch normalztion birnn use tensorlayer?

### ISSUE THREAD

**Issue by guoying1030** [tensorlayer/tensorlayer#86]
how to write batch normalztion birnn use tensorlayer?

**Comment by zsdonghao** [comment]
@guoying1030 do you mean apply batch norm on the RNN output? 
my colleague say it is not good for RNN, you should use layer normalization

**Comment by ruzrobert** [comment]
Hello!
@guoying1030  , @zsdonghao  . Look what I have found today:
https://github.com/tensorflow/tensorflow/issues/1736
"there is an implementation here of batch normalized lstm: https://github.com/OlavHN/bnlstm/blob/master/lstm.py
explained in the article http://olavnymoen.com/2016/07/07/rnn-batch-normalization
inspired from the paper https://arxiv.org/abs/1603.09025"

So it is okay for RNN, but it is different.
Code for that RNN batch normalization is available in https://github.com/OlavHN/bnlstm

**Comment by zsdonghao** [comment]
@ruzrobert 
Thanks, that is useful ~  my colleague is implementing layer normalization. 
Let me mark this issue as `help wanted`.

**Comment by zsdonghao** [comment]
Hi, i close this issue now, feel free to reopen if you want discussion~

To implement different kinds of RNN, we can feed in any third party cell functions into `RNNLayer` and `DynamicRNNLayer`. Other normalisations like layer norm can be found [here]( http://tensorlayer.readthedocs.io/en/latest/modules/layers.html#normalization-layer).
